src/components/trade.component.js

```diff
@@ -0,0 +1,85 @@
+import React, {Component} from 'react';
+import axios from 'axios';
+import TableRow from './TableRow';
+
+const API_URL = 'https://data-fix.smt-data.com/fixprof/instruments/prices';
+const SECONDS = 30;
+
+export default class Trade extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {data: [], response: {}, refreshTime: 30000, currentCount: SECONDS};
+    }
+
+    componentDidMount() {
+        var intervalId = setInterval(this.timer, 1000);
+        this.setState({intervalId: intervalId});
+        this.getTradeData();
+        setInterval(this.getTradeData, this.state.refreshTime);
+    }
+
+    componentWillUnmount = () => {
+        clearInterval(this.state.intervalId);
+    };
+
+    timer = () => {
+        if (this.state.currentCount - 1 < 0) {
+            this.setState({currentCount: SECONDS});
+        } else {
+            this.setState({currentCount: this.state.currentCount - 1});
```

Comment:
this is not wrong, but if you would want to make your code look fancier, you can also do: 
```suggestion
            this.setState(state => ({currentCount: state.currentCount - 1}));
```